test/functional/shared.js

```diff
@@ -89,9 +89,20 @@ function withTempDb(name, options, client, operation, errorHandler) {
 function withClient(client, operation, errorHandler) {
   const cleanup = makeCleanupFn(client);
 
+  function operationHandler(client) {
+    // run the operation
+    const result = operation(client);
+    // if it returns a callback, wrap it in a Promise
+    if (typeof result === 'function') {
```

Comment:
I think we have a [helper](https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/utils.js#L525-L533) for this.